aws/service.go

```diff
@@ -1445,6 +1445,10 @@ func getClientForQuerySupportedRegionWithExclusions(ctx context.Context, d *plug
 		return nil, fmt.Errorf("getClientForQuerySupportedRegion called without a region in QueryData")
 	}
 
+	if region == "global" {
```

Comment:
@karanpopat please add a comment regarding why we need this check for future understanding.